crates/ra_hir_ty/src/expr.rs

```diff
@@ -167,7 +166,14 @@ impl<'a, 'b> ExprValidator<'a, 'b> {
 
         let variant_data = variant_data(db.upcast(), variant_def);
 
-        let lit_fields: FxHashSet<_> = fields.iter().map(|f| &f.name).collect();
+        let lit_fields: FxHashSet<_> = fields
+            .iter()
+            .filter_map(|f| {
+                // TODO: check if cfg_is_enabled with .attrs ?
```

Comment:
SO, if the cfg is disabled, there shouldn't be fields here in the first place, beucase they weren't alloced when we created corresponding VariantData